src/lib/features/project/project-service.test.ts

```diff
@@ -134,31 +132,89 @@ describe('enterprise extension: enable change requests', () => {
             TEST_AUDIT_USER,
             async () => [],
         );
+
+        const { events } = await eventService.getEvents();
+        expect(events[0].data.changeRequestEnvironments).toBeUndefined();
     });
 
     test('the emitted event contains the list of change request envs returned from the extension function, *not* what was passed in', async () => {
-        expect.assertions(1);
-        const service = createService();
-
-        // @ts-expect-error: we want this to pass to test the functionality
-        service.environmentStore.exists = () => true;
+        const { service, environmentStore, eventService } = createService();
+        await environmentStore.create({
+            name: 'dev',
+            type: 'production',
+            enabled: true,
+            protected: false,
+            sortOrder: 0,
+        });
 
         const crEnvs = [{ name: 'prod', requiredApprovals: 10 }];
 
-        const storeEvent = async (e: IBaseEvent) => {
-            expect(e.data.changeRequestEnvironments).toMatchObject(crEnvs);
-        };
+        const projectId = 'fake-project-id';
+        await service.createProject(
+            {
+                id: projectId,
+                name: 'fake-project-name',
+                changeRequestEnvironments: [
+                    { name: 'dev', requiredApprovals: 1 },
+                ],
+            },
+            {
+                id: 5,
+                permissions: [],
+                isAPI: false,
+            },
+            TEST_AUDIT_USER,
+            async () => crEnvs,
+        );
 
-        // @ts-expect-error: for testing purposes
-        service.eventService.storeEvent = storeEvent;
+        const { events } = await eventService.getEvents();
+        expect(events).toMatchObject([
```

Comment:
this is what replaced mock verifications. We check what event was actually stored. I think it's more explicit